Under The Weather/Networking/NetworkService.swift

```diff
@@ -8,65 +8,62 @@
 import Foundation
 
 final class NetworkService: NetworkServiceProtocol {
-    
+
     static let sharedInstance = NetworkService()
-    
-    private let apiKeyObject: APIKeysProvider = APIKeysProvider()
-    internal var urlSession: URLSession
-    private var citiesArray = [(String,Weather)]()
-    
+
+    private let apiKeyObject: APIKeysProvider = .init()
+    var urlSession: URLSession
+    private var citiesArray = [(String, Weather)]()
+
     init(urlSession: URLSession = .shared) {
         self.urlSession = urlSession
     }
-    
+
     func citySearch(city: String, completionHandler: @escaping (Result<[Cities], NetworkError>) -> Void) {
         guard let _ = apiKeyObject.weatherApiKey else {
             completionHandler(.failure(NetworkError.invalidKey))
             return
         }
-        
-        let task = urlSession.dataTask(with: WeatherEndpoint.citySearch(with: city).url) { data, response, error in
-            
+
+        let task = urlSession.dataTask(with: WeatherEndpoint.citySearch(with: city).url) { data, _, error in
+
             guard let data = data, error == nil else {
                 completionHandler(.failure(NetworkError.invalidSearch))
                 return
             }
             do {
                 let response = try
```

Comment:
Isn't response captured from above, but you've replaced it with _

Reply:
`response` isn't used within the closure